src/macros.php

```diff
@@ -297,3 +299,49 @@
         return $this->reverse()->after($currentItem, $fallback);
     });
 }
+
+if (! Collection::hasMacro('paginate')) {
+    /*
+     * Paginate the given collection
+     *
+     * @param int $perPage
+     * @param int $total
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Pagination\LengthAwarePaginator
+     */
+    Collection::macro('paginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, int $total = null, array $options = []): LengthAwarePaginator {
+        $page = $page ?: LengthAwarePaginator::resolveCurrentPage($pageName);
+        $results = $this->forPage($page, $perPage);
+        $total = $total ?: $this->count();
+        $options += [
+            'path' => LengthAwarePaginator::resolveCurrentPath(),
```

Comment:
The `path` key? Because you can do it with other path not for your site. For example, you have website `foo.bar`  and you put `site.com` in `path` key, links method will generate you something like `site.com?page=1`, `site.com?page=2`, and etc. 
Anything else?